phoenix-core/src/main/java/org/apache/phoenix/schema/PTableKey.java

```diff
@@ -26,7 +28,8 @@
     public PTableKey(PName tenantId, String name) {
         Preconditions.checkNotNull(name);
         this.tenantId = tenantId;
-        this.name = name;
+        this.name = !name.contains(QueryConstants.NAMESPACE_SEPARATOR) ? name
```

Comment:
We are still maintaining phoenix table name(A.B) in cache. so that's why if by mistake physical table is passed as a key , it should be resolve correctly. I think I just kept this for precaution only. LocalIndexIT/ViewIndexIT covers it right?